jcl/source/common/JclSysInfo.pas

```diff
@@ -4293,35 +4294,64 @@ function GetWindowsReleaseName: String;
     Result := '';
 end;
 
+function GetWindowsReleaseCode: String;
+var
+  WindowsReleaseId: Integer;
+begin
+  // Looks much like the 'GetWindowsReleaseCodeName', except for the Windows 10 versions
+  // prior to Release Id 1903 - those have a different 'code' vs the 'code name'.
+  if IsWin10 then
+  begin
+    WindowsReleaseId := GetWindowsReleaseId;
+    if WindowsReleaseId < 1903 then
+      Result := IntToStr(WindowsReleaseId)
+    else
+    case WindowsReleaseId of
+      1903:
+        Result := '19H1';
+      1909:
+        Result := '19H2';
+      2004:
+        Result := '20H1';
+      2009:
+        Result := GetWindowsDisplayVersion;
+      else
+        Result := '';
```

Comment:
So this means that if I get a WindowsReleaseId of say, 2105, I will get an empty string back. Looks quite strange to me.

Reply:
Yes, that's because no other ReleaseId's exist at this time.
Starting ReleaseId 2009, MS has stopped increasing the value: for newer versions, only the DisplayVersion is changed...  (see comment in GetWindowsReleaseId/GetWindowsDisplayVersion).

So basically it's at this point unknown, what 'info' is needed to be returned when the ReleaseId changes to something else than 2009.

Also, `GetWindowsReleaseId` is only used with Windows 10.
With Windows 11 (and other versions) only `GetWindowsDisplayVersion` is used.